graphql-ballerina/query_depth_validator.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql.parser;
+
+class QueryDepthValidator{
+    private parser:DocumentNode documentNode;
+    private int queryDepth = 1;
+    private int maxQueryDepth = 0;
+    private int queryDepthLimit;
```

Comment:
What's the difference between the `maxQueryDepth` and the `queryDepthLimit`?

Reply:
maxQueryDepth - maximum depth of the query that received.
queryDepthLimit - User given Limit of Depth that a query can have.